CONTRIBUTING.rst

```diff
@@ -475,38 +586,59 @@ Pull Request Guidelines
 Before you submit a pull request (PR) from your forked repo, check that it meets
 these guidelines:
 
--   Include tests, either as doctests, unit tests, or both, to your pull
-    request.
+-   Include tests, either as doctests, unit tests, or both, to your pull request.
 
     The airflow repo uses `GitHub Actions <https://help.github.com/en/actions>`__ to
     run the tests and `codecov <https://codecov.io/gh/apache/airflow>`__ to track
     coverage. You can set up both for free on your fork. It will help you make sure you do not
     break the build with your PR and that you help increase coverage.
+    Also we advise to install locally `pre-commit hooks <STATIC_CODE_CHECKS.rst#pre-commit-hooks>`__ to
+    apply various checks, code generation and formatting locally at the time you make a local commit - which
+    gives you near-immediate feedback on things you need to fix before you push your code to the PR, or in
+    many case it will even fix it for you locally so that you can add and commit it straight away.
 
--   Follow our project's `Coding style and best practices`_.
-
-    These are things that aren't currently enforced programmatically (either because they are too hard or just
-    not yet done.)
+-   Follow our project's `Coding style and best practices`_. Usually we attempt to enforce the practices by
+    having appropriate pre-commits These are things amongst them that aren't currently enforced
+    programmatically (either because they are too hard or just not yet done).
 
--   `Rebase your fork <http://stackoverflow.com/a/7244456/1110993>`__, and resolve all conflicts.
+-   We prefer that you ``rebase`` your Pull Request (and do it quite often) rather than merge. It leads to
```

Comment:
```suggestion
-   We prefer that you ``rebase`` your pull request (and do it quite often) rather than merge. It leads to
```

nitpick: I saw lower case is used elsewhere. not sure whether we should use upper case here

Reply:
I looked at it and I wil use "Pull Request (PR)" at the beginning of a chaper and later in the same chapter will just use PR. It's commonly used acronym and good practice is to introduce a term with acronym at the beginning and then just use acronmy in each chapter that can be read separately.